CRM/Utils/System.php

```diff
@@ -1416,8 +1416,14 @@ public static function civiExit($status = 0) {
   public static function flushCache() {
     // flush out all cache entries so we can reload new data
     // a bit aggressive, but livable for now
-    $cache = CRM_Utils_Cache::singleton();
-    $cache->flush();
+    CRM_Utils_Cache::singleton()->flush();
+    if (Civi\Core\Container::isContainerBooted()) {
+      Civi::cache('settings')->flush();
+      Civi::cache('js_strings')->flush();
+      Civi::cache('community_messages')->flush();
+      CRM_Extension_System::singleton()->getCache()->flush();
```

Comment:
or is that totally covered here ?

Reply:
I think the extension concern discussed above is addressed more by saying, "Extensions don't use this thing".

The code there does mention extensions (`CRM_Extension_System::singleton()->getCache()->flush();`), but this is metadata used by the extension-system tracking info *about* extensions (specifically the data structure which says "the source code for `foo` is located in `sites/default/files/civicrm/ext/foo`). The line is probably redundant or irrelevant in most usage, but there's arguably an edge-case where it's needed to maintain the behavior of "flush the big yellow circle".

Reply:
@totten I was more thinking about whether I might want to use cache prefixes in an extension - although I guess I can agree that is not something in use now so it could be punted